src/common.rs

```diff
@@ -98,6 +101,19 @@ pub fn invoke<C: FromIterator<Option<I>> + 'static, I>(
             let v = jiter_find(s.as_ref().map(String::as_str), &path).ok();
             Ok(ColumnarValue::Scalar(to_scalar(v)))
         }
+        ColumnarValue::Scalar(ScalarValue::Union(type_id_value, _, _)) => {
+            let opt_json = if let Some((_, value)) = type_id_value {
+                if let ScalarValue::Utf8(s) = value.as_ref() {
+                    s.as_ref()
+                } else {
+                    None
+                }
+            } else {
+                None
+            };
```

Comment:
I think this can simplify:

```suggestion
            let opt_json = if let Some(ScalarValue::Utf8(s))) = type_id_value.map(|(_, value)| value).as_deref() {
               Some(s.as_str())
            } else {
                None
            };
```

Reply:
changed.